pyrefinebio/script.py

```diff
@@ -0,0 +1,183 @@
+import click
+import json
+
+import pyrefinebio.high_level_functions as hlf
+
+from pyrefinebio.exceptions import DownloadError
+
+
+@click.group()
+def cli():
+    """RefineBio CLI"""
+    pass
+
+
+class DictParamType(click.ParamType):
+    name = "dict"
+
+    def convert(self, value, param, ctx):
+        try:
+            return json.loads(value)
+        except:
+            self.fail(
+                "expected valid json string, got: {0}".format(value),
+                param,
+                ctx
+            )
+
+
+class ListParamType(click.ParamType):
+    name = "list"
+
+    def convert(self, value, param, ctx):
+        try:
+            return value.split()
+        except:
+            self.fail(
+                "expected valid string in list form, got: {0}".format(value),
+                param,
+                ctx
+            )
+
+
+@cli.command()
+@click.argument("entity", nargs=1)
+def help(entity=None):
+    """
+    Prints out information about pyrefinebio's classes and functions.
+
+    ENTITY is the pyrefinebio class or function you want information about.
+
+    To get information about a class method, pass in the class name and method name
+    separated by either a space or a `.`
+
+    Example:
+
+    $ refinebio help "Sample.search"
+    """
```

Comment:
Does this get set as the helpstring by default?

Reply:
I just added `no_args_is_help=True` to this click command

now when you specify no args to `refinebio help` you get this:

![image](https://user-images.githubusercontent.com/52019096/107659458-2b55df00-6c55-11eb-9467-f30596dd994b.png)

lmk if this looks good to you

Reply:
Ok sweet, this does look pretty good. However if I'm using this from the command line I'd actually probably want to know what my choices of sub-commands are, then I might see how to call the one I'm interested in by calling `refinebio help <MYCOMMAND>`.

Reply:
ohh ok I think you get that by doing `$ refinebio --help` or just `$ refinebio`

![image](https://user-images.githubusercontent.com/52019096/107664996-fe0c2f80-6c5a-11eb-85fa-93161a2f76a5.png)

Then you could do `$ refinebio COMMAND --help` to get help for that command.

This help command really only gets help for the actual classes and functions in pyrefinebio. If you wanted to use it to get help for the cli commands you would have to do `$ refinebio help "script.dowload_dataset"` and in that case it only prints the docstring.

![image](https://user-images.githubusercontent.com/52019096/107664777-c604ec80-6c5a-11eb-886f-42da624c96ad.png)

![image](https://user-images.githubusercontent.com/52019096/107664898-dfa63400-6c5a-11eb-9288-6addb186b270.png)

Reply:
Ah ok, that makes sense then. Is there any way to have `refinebio help` do the same thing as `refinebio --help`? If I type the first one and it outputs, I might not think to try the second.